### PR TITLE
Docs: fixing links and linkcheck on Github action

### PR DESCRIPTION
**Pull request by unkcpz**
hi @chrisjsewell, the modifications of https://github.com/aiidateam/aiida-core/pull/5001 are included here with linkcheck.

### CODE REVIEW

**Comment by unkcpz** [comment]
Here is a problem with this github action, when the action failed, it doesn't give verbose information at last, so it is hard to find from the action page which link is broken. Have to either view the file from the GitHub page or build locally and search for the keyword 'broken'. 
I currently put the action in test ci for every push and pull request, I believe it might be better to only test this in the new release phase.

**Comment by ltalirz** [comment]
Thanks a lot for this @unkcpz !

Since this turned up quite a number of broken links in the howto section (notifying also @sudarshanv01 ), I think we should get at least the fixes merged today.

> it doesn't give verbose information at last, so it is hard to find from the action page which link is broken. Have to either view the file from the GitHub page or build locally and search for the keyword 'broken'.

This can be done on github actions I believe, i.e. store the output of the action and in the next step fail if `broken` was detected.


> I currently put the action in test ci for every push and pull request, I believe it might be better to only test this in the new release phase.

I guess it would only be needed if a PR changes the docs?
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths
If we limit it to that I think we can run it always

I'll have a quick go at this.

**Comment by ltalirz** [comment]
Or @unkcpz if you have the time to have a look that would be great as well - I will come back to this in 2h (need to work on something else first)

**Comment by chrisjsewell** [comment]
> I guess it would only be needed if a PR changes the docs

For that it would need to go in a separate workflow file, e.g. for .github/workflows/ci-code.yml it is skipped if there are only document changes.

Note obviously this would not pick up any broken links added in docstrings, for a non-doc PR

**Comment by unkcpz** [comment]
Thanks! @ltalirz @chrisjsewell 
The action is put in a separate file now and the outputs are grepped to give the results. 

It's true in this case the changes in docstrings will not trigger the action in the pull request, but I believe this is rare and can be left on `push` phase.

File: aiida/tools/data/orbital/realhydrogen.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ class RealhydrogenOrbital(Orbital):
     OrbitalData class.
 
     Following the notation of table 3.1, 3.2 of Wannier90 user guide
-    http://www.wannier.org/doc/user_guide.pdf
+    (which can be download from http://www.wannier.org/support/)
```
Comment (ltalirz) [inline aiida/tools/data/orbital/realhydrogen.py:261]:
```suggestion
    (which can be downloaded from http://www.wannier.org/support/)
```

File: docs/source/howto/plugins_develop.rst [hunk 2]
```diff
@@ -18,22 +18,22 @@ Creating a plugin package
...
+AiiDA plugins can be bundled and distributed in a `Python package <packages_>`_ that provides a set of extensions to AiiDA.
 
 .. note::
 
   The Python community uses the term 'package' rather loosely.
-  Depending on context, it may refer simply to a folder containing individual Python modules or it may include the files necessary for building and installing a package to be distributed via the `Python Package Index (PyPI) <pypi>`_.
+  Depending on context, it may refer simply to a folder containing individual Python modules or it may include the files necessary for building and installing a package to be distributed via the `Python Package Index (PyPI) <pypi_>`_.
 
-.. _packages: https://docs.python.org/2/tutorial/modules.html?highlight=package#packages
+.. _packages: https://docs.python.org/3/tutorial/modules.html?highlight=package#packages
```
Comment (ltalirz) [inline docs/source/howto/plugins_develop.rst:28]:
```suggestion
.. _packages: https://docs.python.org/3/tutorial/modules.html#packages
```

**Review by ltalirz — CHANGES_REQUESTED** [review]
thanks a lot @unkcpz for this, this come just at the right time :-)

I have just two minor change requests, then I suggest we get this merged.

Please let me know in the next 15 minutes whether you're going to work on finishing this - otherwise I can take over (I know it' s quite late where you are by now ;-) )

**Comment by ltalirz** [comment]
@unkcpz about the broken link to aiida-diff - don' get crazy.
You can either add it to the ignore list or remove it for being too specific

I suspect that github is doing a special processing of the "anchor"  value that is not what the sphinx checker expects

**Comment by ltalirz** [comment]
As mentioned above, since it's late where you are, let me know if you're going to finish this

**Comment by unkcpz** [comment]
Hi @ltalirz, I have to go now, please make changes to finish this, if necessary.

**Review by ltalirz — APPROVED** [review]

**Comment by ltalirz** [comment]
mentioning @csadorf for info (this is not about dependencies but about the github workflow setup)

I've merged this now since it contains important fixes for the tutorial.
If any modifications should be made to setup of the github workflow that can be done in a follow-up PR

**Pull request merged**